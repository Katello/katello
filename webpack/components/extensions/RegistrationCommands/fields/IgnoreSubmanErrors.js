import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'foremanReact/common/helpers';

import { FormGroup, Checkbox } from '@patternfly/react-core';
import LabelIcon from 'foremanReact/components/common/LabelIcon';
import { translate as __ } from 'foremanReact/common/I18n';

const IgnoreSubmanErrors = ({ value, onChange, isLoading }) => (
  <FormGroup fieldId="reg_katello_ignore">
    <Checkbox
      label={
        <span>
          {__('Ignore errors')}{' '}
          <LabelIcon text={__('Ignore subscription manager errors')} />
        </span>
      }
      id="reg_katello_ignore"
      onChange={() => onChange({ ignoreSubmanErrors: !value })}
      isDisabled={isLoading}
      isChecked={value}
    />
  </FormGroup>
);

IgnoreSubmanErrors.propTypes = {
  value: PropTypes.bool,
  onChange: PropTypes.func,
  isLoading: PropTypes.bool,
};

IgnoreSubmanErrors.defaultProps = {
  value: false,
  onChange: noop,
  isLoading: false,
};

export default IgnoreSubmanErrors;
