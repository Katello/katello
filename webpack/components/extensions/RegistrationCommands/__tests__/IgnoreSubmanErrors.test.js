import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
import IgnoreSubmanErrors from '../fields/IgnoreSubmanErrors';

const fixtures = {
  renders: { value: false, isLoading: false, onChange: () => {} },
};

describe('IgnoreSubmanErrors', () =>
  testComponentSnapshotsWithFixtures(IgnoreSubmanErrors, fixtures));
