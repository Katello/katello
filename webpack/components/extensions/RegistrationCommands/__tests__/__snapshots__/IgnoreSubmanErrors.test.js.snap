// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IgnoreSubmanErrors renders 1`] = `
<FormGroup
  fieldId="reg_katello_ignore"
>
  <Checkbox
    className=""
    id="reg_katello_ignore"
    isChecked={false}
    isDisabled={false}
    isLabelWrapped={false}
    isRequired={false}
    isValid={true}
    label={
      <span>
        Ignore errors
         
        <LabelIcon
          text="Ignore subscription manager errors"
        />
      </span>
    }
    onChange={[Function]}
    ouiaId="reg-katello-ignore"
    ouiaSafe={true}
  />
</FormGroup>
`;
